cpp/include/cudf_test/debug_utilities.hpp

```diff
@@ -20,8 +20,9 @@
 // Do NOT include any header that directly or indirectly depends on gtest.
 // It is to make sure this file can be used and compiled outside of the test framework.
 //
-// In order to use the APIs in this file outside of the test framework, simply include the source
-// file `test/utilities/debug_utilities.cu` into the list of source files of the target build.
+// In order to use the APIs listed below outside of the test framework, simply include this header
+// and add the corresponding source file `test/utilities/debug_utilities.cu` into the CMake
+// target build.
```

Comment:
Would it be possible to include the file as a ifdef in the CMakeLists.txt so it could be compiled in with a cmake parameter?

Reply:
The source file ` tests/utilities/debug_utilities.cu` is already part of the cudftestutil. We shouldn't be building it part of two different DSO as we could run into duplicate symbol issues from calling tests.

All the libcudf tests and benchmarks have access to this function since they link to to `cudftestutil`. If you are debugging something external, you can gain access to this feature by adding `cudftestutil` to your link line and not be required to recompile libcudf.

Reply:
Oh I only see your comment after pushed my cmake changes :smile: . Anyway, I tried to avoid duplicate symbols using `if/else`. Please take a look.

Reply:
In addition, I don't know if it is possible to add `cudftestutil` into `libcudf.so`, not the exec application, since I need to output `cudf::test::print` into `libcudf.so` to be loaded in Spark.

Reply:
spark builds libcudf statically right and embeds some extra code into a new `libcudf.so` no reason that can't also embed `cudftestuil`. 

This approach is the wrong way to accomplish our goals.

Reply:
Okay, so I reverted the approach with cmake changes.
Personally I had (and still have) so many trouble with linking `cudftestutil` so I still prefer manually add the source file. In addition, extracting the `print` module should also make the code base cleaner.

Reply:
In fact, adding `cudftestutil` is not very straightforward, and I don't know how to fix this:
```
[INFO]      [exec] /opt/rh/devtoolset-11/root/usr/libexec/gcc/x86_64-redhat-linux/11/ld: /home/nghiat/Devel/jni/target/libcudf-install/lib64/libgtest.a(gtest-all.cc.o): relocation R_X86_64_32S against symbol `_ZTVN7testing8internal17TestEventRepeaterE' can not be used when making a shared object; recompile with -fPIC
```

Reply:
You need to enable position independent code ( the -fPIC flag )for the `libgtest ` to resolve those link error. Can you please open this as an issue on `rapids-cmake` and I can fix the issue.